compiler/include/iterator.h

```diff
@@ -85,6 +85,7 @@ void gatherPrimIRFieldValByFormal();
 void cleanupPrimIRFieldValByFormal();
 
 void lowerIterator(FnSymbol* fn);
+void lowerThunk(FnSymbol* fn);
```

Comment:
To mee, `resolution.h` is a misleading place since this occurs in pass 20, lowerIterators. I think it makes sense to leve `lowerThunk` near `lowerIterator` since they occur in the same pass and behave very similarly. I'm also open to other places -- or other interpretations of "resolution.h" :)

Reply:
The more I see problems like this, the more I want to have `thunk.h`. Or if I was the king of the world, I would just make this logic its own pass.

I am not proud of the pattern in our compiler code where we stick things related or unrelated next to things that happen right before or after them. That's another thing that I've been doing but trying to steer away from as much as I can these days. This particular case looks like it could be easily handled in a way to avoid that. If you agree with that, that is.

I don't intend to try to convince you. My current preference is to have a `thunk.h`. But I can go with what your final call is.

Reply:
I've added a `thunks.h` and moved all implementation code from all passes into `thunks.cpp`. I also took that opportunity to add pass separators to `thunks.cpp`. I think it came out rather well.

Reply:
Are we past the point of not adding new passes because of the compiler performance tracking challenges? Probably not. I am sorry if I led you down that path.

Reply:
I didn't make a new pass (in fact, I don't think that I can, but only due to pragmatic concerns). I haven't thought about the general possibility.

Reply:
Oh, OK. I misunderstood, then. Thanks for those changes.